### PR TITLE
LPS-127425 Parallelize module requests when JSModuleLauncher receives an array

### PR DESCRIPTION
**Pull request by izaera**
This is an optimization we had pending in the webpack epic. Because I'm waiting for https://github.com/liferay-frontend/liferay-portal/pull/839 to be merged and this is so isolated, I'm sending it.

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-module-launcher/src/main/java/com/liferay/frontend/js/module/launcher/internal/JSModuleLauncherImpl.java [hunk 1]
```diff
@@ -129,27 +130,55 @@ private String _getScriptBody(
...
+			javascriptSB = new StringBundler(
+				6 + (5 * jsModuleDependencies.size()));
 
-		javascriptSB.append(javaScriptCode);
+			javascriptSB.append("(function() {Promise.all([");
 
-		for (int i = 0; i < jsModuleDependencies.size(); i++) {
-			javascriptSB.append("});");
-		}
+			for (JSModuleDependency jsModuleDependency : jsModuleDependencies) {
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-module-launcher/src/main/java/com/liferay/frontend/js/module/launcher/internal/JSModuleLauncherImpl.java:164]: The obvious question: how many of these dependencies de we expect to be?

Comment (izaera) [reply modules/apps/frontend-js/frontend-js-module-launcher/src/main/java/com/liferay/frontend/js/module/launcher/internal/JSModuleLauncherImpl.java:164]: Not too many. As far as I've seen it shouldn't be more than three (the react renderer + react component + props transformer).

However, if someone asks for 100 modules it's their fault, not  ours :-D

Comment (izaera) [reply modules/apps/frontend-js/frontend-js-module-launcher/src/main/java/com/liferay/frontend/js/module/launcher/internal/JSModuleLauncherImpl.java:164]: Think of this method as `open()` to open files: if you open too many files, it's not `open()`'s fault :shrug:

Comment (wincent) [reply modules/apps/frontend-js/frontend-js-module-launcher/src/main/java/com/liferay/frontend/js/module/launcher/internal/JSModuleLauncherImpl.java:164]: 3 sounds like a good number.

And if and when somebody asks for 100 we can pool/batch.

**Review by wincent — APPROVED** [review]

**Comment by izaera** [comment]
ci:forward

**Pull request closed without merging**